Sources/Conversion.swift

```diff
@@ -0,0 +1,233 @@
+                        lhs[key] = value
+                    }
+                    
+                    return lhs
+                    } as [String: Node]
+                
+                return .object(object)
+            }
+        case let objectId as ObjectId:
+            return objectId.makeNode(in: nil)
+        case let regex as RegularExpression:
+            return regex.makeNode(in: nil)
+        case let date as Date:
+            return .date(date)
+        case let binary as Binary:
+            return binary.makeNode(in: nil)
+        case let date as Date:
+            return .date(date)
+        default:
+            return .null
+        }
+    }
+}
+
+extension Binary : NodeConvertible {
+    /// Makes a Node from Binary
+    ///
+    /// TypeIdentifier metadata gets lost
+    public func makeNode(in context: Context?) -> Node {
+        return .bytes([UInt8](self.data))
+    }
+    
+    /// Makes a Binary blob from Node.binary
+    ///
+    /// TypeIdentifier metadata is lost
+    public init(node: Node) throws {
+        guard let bytes = node.bytes else {
+            throw NodeError.unableToConvert(input: node, expectation: "\(Bytes.self)", path: [])
+        }
+        
+        self = Binary(data: bytes, withSubtype: .generic)
+    }
+}
+
+extension ObjectId : NodeConvertible {
+    /// Converts the hexString to a Node.string but prefixes the hexstring with `oid:`
+    public func makeNode(in context: Context?) -> Node {
+        return Node(.string("oid:" + self.hexString))
+    }
+    
+    /// Checks for and removed the `oid:` prefix and converts the hexString to an Objectid
+    public init(node: Node) throws {
+        guard var string = node.string, string.hasPrefix("oid:") else {
+            throw NodeError.unableToConvert(input: node, expectation: "\(ObjectId.self)", path: [])
+        }
+        
+        string.characters.removeFirst(4)
+        self = try ObjectId(string)
+    }
+}
+
+extension RegularExpression : NodeConvertible {
+    /// RegularExpression has the regex string as well as options. So we're using an array for the data here.
+    ///
+    /// To identify this as a regex, the String `"regex"` will be used as the first element.
+    ///
+    /// Using this pattern, convert RegularExpression to a Node
+    public func makeNode(in context: Context?) -> Node {
+        return Node(.array([.string("regex"), .string(self.pattern), .number(.uint(self.options.rawValue))]))
+    }
+    
+    
+    /// RegularExpression has the regex string as well as options. So we're using an array for the data here.
+    ///
+    /// To identify this as a regex, the String `"regex"` will be used as the first element
+    ///
+    /// Using this pattern, try to convert Node to a RegularExpression
+    public init(node: Node) throws {
+        guard let array = node.array, array.count == 3, array[0].string == "regex", let pattern = array[1].string, let options = array[2].uint else {
```

Comment:
```swift
guard 
    let array = node.array, array.count == 3, 
    array[0].string == "regex", 
    let pattern = array[1].string, 
    let options = array[2].uint 
else
```

formatting like this can make it a lot more readable for narrow widths